src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -983,59 +983,130 @@ private RuntimeExecutionResults ProcessCore()
+                    
+                    // Retrieve processor options
                     try
                     {
-                        /*
-                         * TODO (#214): for now, the engine ignores the groups returned by processing sources that implement
-                         * IDataSourceGrouper. In the future, this needs to
-                         * 1. Check if the processing source implements IDataSourceGrouper (which it must after V2)
-                         * 2. If not, do what we're doing now and just create a new DataSourceGroup with the default
-                         *     processing mode
-                         * 3. If yes, use an IDataSourceGroupResolver to pick out which of the IDataSourceGroup(s)
-                         *    returned by the processing source should be used for processing.
-                         *      a. If the engine user does not supply a resolver, a sensible one should be used. E.g.
-                         *         one that picks groups that maximize the number of data sources processed without a
-                         *         data source being processed in more than 1 group
-                         * 4. Add a new executor for each IDataSourceGroups picked out in the above step
-                         */
-                        var executionContext = new SDK.Runtime.ExecutionContext(
-                            new DataProcessorProgress(),
-                            x => ConsoleLogger.Create(x.GetType()),
-                            processingSource,
-                            new DataSourceGroup(dataSources, new DefaultProcessingMode()), // see above TODO
-                            processingSource.Instance.MetadataTables,
-                            new RuntimeProcessorEnvironment(this.Extensions, this.compositeCookers, this.CreateLogger),
-                            ProcessorOptions.Default);
-
-                        var executor = new ProcessingSourceExecutor();
-                        executor.InitializeCustomDataProcessor(executionContext);
-
-                        executors.Add(executor);
+                        processorOptions = optionsResolver.GetProcessorOptions(processingSource.Guid, dsg);
                     }
                     catch (Exception e)
                     {
+                        optionsFailure = true;
                         this.logger.Error(
-                           "Error processing dataSources {0} on processing source {1}: {2}",
-                           dataSources.ToString(),
+                           "Unable to get processor options for processing source {0} and data sources {1}: {2}",
                            processingSource.Name,
+                           dsg.DataSources.ToString(),
                            e);
                     }
+
+                    // Validate processor options
+                    Debug.Assert(processorOptions != null, "How did we get here");
```

Comment:
We can easily get here if the user-implemented resolver returns null.  Perhaps we should just check for null in `AreUnsupportedOptions`?

Reply:
I added this assert to double check that options = null is what the user expected since it's a unique case that likely shouldn't be hit. We have the `AreUnsupportedOptions` check logged as well. I can remove this line. Thoughts?

Reply:
I would say remove it for two reasons:
1. Debug asserts are more for the developers of the code being asserted, which wouldn't be plugin authors. They should signify an invariant of a system has been broken. Because a resolver can always return null, it is not an invariant that we don't have a null value here
2. If `AreUnsupportedOptions` logs an error if the options are null, we'll terminate processing anyway and the user will see an error anyway

Reply:
Adding a logger.warn instead and will continue since technically I suppose null processor options could be supported.

Reply:
Hmm, what are your thoughts on documenting `IProcessorOptionsResolver.GetProcessorOptions` may not return null? This way we abort processing early if the implementation is out of spec. Processing with this as null might cause headaches in the future